doc/reported_states_during_deployment.md

```diff
@@ -0,0 +1,353 @@
+have come into the correct state.
+
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" == .properties.reported.lastDesiredVerison && 
+.properties.reported.lastDesiredStatus.code == 200 && 
+.properties.desired.systemModules.edgeHub.status == .properties.reported.systemModules.edgeHub.status &&
+```
+
+For each desired module:
+```
+   .properties.desired.modules.<module>.status == .properties.reported.modules.<module>.runtimeStatus &&  
+   .properties.desired.modules.<module>.settings.image == .properties.reported.modules.<module>.settings.image 
+```
+ 
+
+### Deployment is invalid
+
+The IoT Edge runtime has received a deployment, but the deployment was invalid.
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" == .properties.reported.lastDesiredVerison && 
+.properties.reported.lastDesiredStatus.code == 400 or 412  
+```
+
+NOTE: We wouldn’t expect to see “406” unless \$edgeAgent shuts down, or “417” (empty deployment) 
+once the first deployment is applied. These may occur if \$edgeAgent is being updated.
+
+
+### Deployment succeeded, module failed after launch. 
+
+In this state, the IoT Edge runtime has successfuly deployed all modules, but the modules went into
+a incorrect state after being deployed.
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" == .properties.reported.lastDesiredVerison && 
+.properties.reported.lastDesiredStatus.code == 200 && 
+```
+
+For any desired module:
+```
+   .properties.desired.modules.<module>.status  != .properties.reported.modules.<module>.runtimeStatus &&  
+   .properties.desired.modules.<module>.settings.image == .properties.reported.modules.<module>.settings.image 
+```
+
+NOTE: Modules may go into a failed state at any time for any number of reasons, it is up to the 
+user's discretion to determine the appropriate amount of time to wait to determine if a module has 
+not been successfully deployed. 
+
+### Deployment in progress/Deployment failed.
+
+One of the steps that the IoT Edge runtime attempted to apply the current desired state failed. 
+There may be many reasons for this, such as an image pull request is taking too long, the module 
+specification references a bad image, or the container runtime failed for any number of reasons.
+
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" == .properties.reported.lastDesiredVerison && 
+.properties.reported.lastDesiredStatus.code == 500 && 
+```
+
+For any desired module:   
+```
+   .properties.desired.modules.<module>.settings.image != .properties.reported.modules.<module>.settings.image 
+```
+
+NOTE: if this is a new module in the deployment, it will not exist in the reported properties until 
+the module is created.
+
+NOTE: If the module was previously marked as “running” it should stay “running” using the previous 
```

Comment:
Shouldn't this be -  NOTE: If the module was previously marked as “running” it _will_ stay “running” using the previous image until the pull for the new module is complete.

Reply:
I can't guarantee that this old module won't have some issue with the new deployment and crash. I'm hedging bets here, but it's probably OK to say it will, because that is how it is designed to work.